app/models/info_request.rb

```diff
@@ -1424,19 +1424,35 @@ def update_counter_cache(body = public_body)
      PublicBody.set_callback(:save, :after, :purge_in_cache)
   end
 
+  def similar_cache_key
+    "request/similar/#{id}"
+  end
+
   # Get requests that have similar important terms
   def similar_requests(limit=10)
-    xapian_similar = nil
-    xapian_similar_more = false
-    begin
-      xapian_similar = ActsAsXapian::Similar.new([InfoRequestEvent],
-                                                 info_request_events,
-                                                 :limit => limit,
-                                                 :collapse_by_prefix => 'request_collapse')
-      xapian_similar_more = (xapian_similar.matches_estimated > limit)
-    rescue
+    ids, more = similar_ids(limit)
+    [InfoRequest.includes(:public_body => :translations).find(ids), more]
+  end
+
+  # Get the ids of similar requests, and whether there are more
+  def similar_ids(limit=10)
```

Comment:
I deployed this commit to WDTK at `2017-09-01 09-54-01` (https://github.com/mysociety/alaveteli/tree/wdtk-4165-reduce-cache-size + https://github.com/mysociety/whatdotheyknow-theme/tree/reduce-cache-size) to have a look at the performance impact while reviewing.

It looks like new requests are getting initially cached with the default empty values.

```ruby
r = InfoRequest.last
r.similar_ids
# => [[], false]
```

Forcing a recache results in the correct data being cached…

```ruby
Rails.cache.delete "request/similar/#{r.id}"
# => true

r.similar_ids
# Xapian query (0.55564s) Similar: #<InfoRequestEvent::ActiveRecord_Associations_CollectionProxy:0x007fbc4b6b1598>
# => [[26731, 218790, 416415, 352428, 1793, 301704, 49728, 372920, 387908, 10178], true]

r.similar_ids
# => [[26731, 218790, 416415, 352428, 1793, 301704, 49728, 372920, 387908, 10178], true]
```